ignore: ignore

ulfalizer opened the pull request:
Hunting for selects with unsatisfied deps.

(Don't review this PR. I'm cherry-picking fixes into it just for testing. There's separate PRs for fixes that are ready.)

ulfalizer committed:
ignore: ignore

Hunting for selects with unsatisfied deps.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
shell: kconfig: Have SHELL_BACKEND_RTT depend on USE_SEGGER_RTT

Have SHELL_BACKEND_RTT depend on USE_SEGGER_RTT instead of selecting it,
to fix some selects of USE_SEGGER_RTT with unsatisfied dependencies.
USE_SEGGER_RTT was being forced on without checking HAS_SEGGER_RTT.

This means configuration files now have to enable both USE_SEGGER_RTT
and SHELL_BACKEND_RTT to enable SHELL_BACKEND_RTT. At least
samples/subsys/shell/shell_module/prj_minimal_rtt.conf does.

(The original version of this commit added a dependency on
HAS_SEGGER_RTT to SHELL_BACKEND_RTT instead, but Krzysztof Chruscinski
suggested depending on USE_SEGGER_RTT instead.)

Footnote: 'depends on' conditions get propagated to properties, so this
change is the same as adding '... if USE_SEGGER_RTT' to all properties
(and the prompt).

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

nordic-krch commented on subsys/shell/Kconfig.backends:
```diff
@@ -117,6 +117,7 @@ config SHELL_BACKEND_RTT
 	select USE_SEGGER_RTT
```
if `depends on HAS_SEGGER_RTT` is removed then this should be changed to `depends on USE_SEGGER_RTT`. This option should appear only if rtt is used.

ulfalizer replied:
Changed.

Wasn't sure if there might be configuration files setting only `SHELL_BACKEND_RTT` without setting `USE_SEGGER_RTT`. None in-tree at least.

Please review at https://github.com/zephyrproject-rtos/zephyr/pull/22075 btw. This PR is just a dumping ground for cherry-picked fixes while testing.

ulfalizer committed:
console: kconfig: Have CONSOLE_HANDLER dep. on SERIAL_SUPPORT_INTERRUPT

CONSOLE_HANDLER was unconditionally selecting UART_INTERRUPT_DRIVEN (and
depends on it in the code), but UART_INTERRUPT_DRIVEN depends on
SERIAL && SERIAL_SUPPORT_INTERRUPT.

Add a SERIAL_SUPPORT_INTERRUPT dependency to CONSOLE_HANDLER.
CONSOLE_HANDLER already depends on UART_CONSOLE, which depends on
SERIAL, so a SERIAL dependency does not need to be added.

This avoids some selects with unsatisfied dependencies in CI.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
console: Have CONSOLE_GET{CHAR,LINE} dep. on SERIAL_SUPPORT_INTERRUPT

Overlooked dependency in commit 97de99adca ("console: kconfig: Have
CONSOLE_{GETCHAR,GETLINE} depend on UART_CONSOLE"). CONSOLE_GETCHAR and
CONSOLE_GETLINE select CONSOLE_HANDLER, which also depends on
SERIAL_SUPPORT_INTERRUPT.

Fixed some selects with unsatisfied dependencies in CI.

(CONSOLE_HANDLER also depends on SERIAL, but it's redundant, since it
already depends on UART_CONSOLE, which depends on SERIAL. This is
simplified in https://github.com/zephyrproject-rtos/zephyr/pull/22116.)

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
Should all be fixed now.

https://github.com/zephyrproject-rtos/zephyr/pull/22253 removes the whitelisting.
